src/main/java/com/defold/extension/pipeline/TealTranspiler.java

```diff
@@ -97,4 +102,76 @@ public List<Issue> transpile(File pluginDir, File sourceDir, File outputDir) thr
         }
         return result;
     }
+
+    private static void cleanOutput(File sourceDir, File outputDir) throws IOException {
+        if (!outputDir.isDirectory()) {
+            return;
+        }
+
+        // Cyan will only re-transpile a source file if it is newer than the
+        // output file. This is an issue for us due to how the editor prepares
+        // source files for compilation. As a workaround, we will clean the
+        // output directory before invoking Cyan. However, Cyan can be
+        // configured to selectively prune files from the output directory, so
+        // even though we supply the --prune flag, simply emptying the output
+        // directory is too intrusive. Instead, we specifically remove only the
+        // expected output files from the output directory.
+        // See https://github.com/defold/defold/issues/8928 for details.
+        Path sourceDirPath = sourceDir.toPath().toRealPath(LinkOption.NOFOLLOW_LINKS);
+        Path outputDirPath = outputDir.toPath().toRealPath(LinkOption.NOFOLLOW_LINKS);
+        List<Path> expectedOutputFilePaths = getExpectedOutputFilePaths(sourceDirPath, outputDirPath);
+
+        for (Path path : expectedOutputFilePaths) {
+            if (Files.isRegularFile(path, LinkOption.NOFOLLOW_LINKS)) {
+                Files.delete(path);
+            }
+        }
+    }
+
+    private static List<Path> getExpectedOutputFilePaths(Path sourceDirPath, Path outputDirPath) throws IOException {
+        if (sourceDirPath.equals(outputDirPath)) {
+            // Cyan does not write any .lua files in this case.
+            return List.of();
+        }
+
+        List<Path> tealSourceFilePaths = getTealSourceFilePaths(sourceDirPath);
+        return getLuaOutputFilePaths(tealSourceFilePaths, sourceDirPath, outputDirPath);
+    }
+
+    private static List<Path> getTealSourceFilePaths(Path sourceDirPath) throws IOException {
+        PathMatcher tealSourceFilePathMatcher = FileSystems.getDefault().getPathMatcher("glob:**.tl");
+        Path internalDirPath = sourceDirPath.resolve("internal");
```

Comment:
Shouldn't it be `.internal` instead, since this is the name that the editor uses? And in that case, do we even need it, since `.internal` is covered by `Files.isHidden`? Or maybe it's needed because `Files.isHidden` matches dot-files only on linux and mac, but not windows...

Reply:
Ah, nice catch. This is wrong. I intended to exclude the `build` directory, which is not hidden. I'll fix!